### PR TITLE
[browser][MT] enable timer tests

### PR DESCRIPTION
**Pull request by pavelsavara**
after we switched to portable timer in https://github.com/dotnet/runtime/pull/94180

### CODE REVIEW

File: src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs [hunk 1]
```diff
@@ -122,6 +122,7 @@ public static bool IsPrivilegedProcess
 
         public static bool IsThreadingSupported => (!IsWasi && !IsBrowser) || IsWasmThreadingSupported;
         public static bool IsWasmThreadingSupported => IsBrowser && IsEnvironmentVariableTrue("IsBrowserThreadingSupported");
+        public static bool IsNotWasmThreadingSupported => !IsWasmThreadingSupported;
```
Comment (ilonatommy) [inline src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs:125]: You probably did it to make `WasmThreadingSupported` search match both variables, but can we consider `IsWasmThreadingUnsupported` ot `IsWasmThreadingNotSupported`? They make me less confused when reading

**Review by ilonatommy — APPROVED** [review]

**Comment by pavelsavara** [comment]
There are known CI issues listed above.
The tests enabled here passed [Log](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-94217-merge-758f477da37344efba/WasmTestOnBrowser-System.Threading.Timer.Tests/1/console.55ebc356.log?helixlogtype=result)

I opened new issue https://github.com/dotnet/runtime/issues/94253 for
```
[18:04:29] info: [2023-10-31T18:04:29.045Z] [FAIL] System.Runtime.InteropServices.JavaScript.Tests.MarshalTests.MarshalFailedAsynchronousTask
[18:04:29] info: JavasScript promise did not completed.
[18:04:29] info:    at System.Runtime.InteropServices.JavaScript.Tests.MarshalTests.MarshalTask(String helperMethodName, String helperMethodArgs, String resolvedBody)
[18:04:29] info:    at System.Runtime.InteropServices.JavaScript.Tests.MarshalTests.MarshalFailedAsynchronousTask()
[18:04:29] info: --- End of stack trace from previous location ---
```

And
```
[18:04:29] info: JavasScript promise did not completed.
[18:04:29] info:    at System.Runtime.InteropServices.JavaScript.Tests.MarshalTests.MarshalTask(String helperMethodName, String helperMethodArgs, String resolvedBody)
[18:04:29] info:    at System.Runtime.InteropServices.JavaScript.Tests.MarshalTests.MarshalAsynchronousTask()
[18:04:29] info: --- End of stack trace from previous location ---

```

[Log ](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-94217-merge-758f477da37344efba/WasmTestOnBrowser-System.Runtime.InteropServices.JavaScript.Legacy.Tests/1/console.2bef0600.log?helixlogtype=result)

**Pull request merged**